### PR TITLE
METRON-1993: Stellar REST_GET should handle responses when content length is less than zero

### PR DESCRIPTION
**Pull request by merrimanr**
## Contributor Comments
This PR removes the check on the response content length header.  From what I can tell, this header is optional in the HTTP/2 specification:  https://tools.ietf.org/html/rfc7540#section-8.1.2.6.

This can be tested by using the `REST_GET` function with an endpoint that doesn't include the content length header.  Before an empty response was returned:
```
[Stellar]>>> REST_GET('https://pokeapi.co/api/v2/pokemon')
[Stellar]>>> 
```
After this change it should return a response with json content.

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  

In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel).
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

### For code changes:
- [x] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [x] Have you included steps or a guide to how the change may be verified and tested manually?
- [x] Have you ensured that the full suite of tests and checks have been executed in the root metron folder via:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```

- [x] Have you written or updated unit tests and or integration tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered by building and verifying the site-book? If not then run the following commands and the verify changes via `site-book/target/site/index.html`:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.

### CODE REVIEW

**Comment by nickwallen** [comment]
Was there enough logging to debug this issue originally?  Can we add debug logs so it is a little easier to troubleshoot what is happening under the covers?

**Comment by merrimanr** [comment]
The latest commit adds a debug line with a summary of the request/response.  That should make it easier to troubleshoot issues like this in the future.

File: metron-stellar/stellar-common/src/test/java/org/apache/metron/stellar/dsl/functions/RestFunctionsTest.java [hunk 1]
```diff
@@ -602,4 +606,27 @@ public void restGetShouldCloseHttpClient() throws Exception {
     verifyNoMoreInteractions(httpClient);
   }
 
+  @Test
+  public void restGetShouldParseResponse() throws Exception {
```
Comment (justinleet) [inline metron-stellar/stellar-common/src/test/java/org/apache/metron/stellar/dsl/functions/RestFunctionsTest.java]: Do we want these to be separate tests? it'd be nice to know exactly what each failure is if we ever have them in the future.

Comment (merrimanr) [reply metron-stellar/stellar-common/src/test/java/org/apache/metron/stellar/dsl/functions/RestFunctionsTest.java]: No problem.  Done.

File: metron-stellar/stellar-common/src/main/java/org/apache/metron/stellar/dsl/functions/RestFunctions.java [hunk 2]
```diff
@@ -244,15 +244,13 @@ protected Object doGet(RestConfig restConfig, HttpGet httpGet, HttpClientContext
         scheduledFuture.cancel(true);
       }
       int statusCode = response.getStatusLine().getStatusCode();
+      LOG.error("request = {}; response = {}", httpGet, response);
```
Comment (mmiklavc) [inline metron-stellar/stellar-common/src/main/java/org/apache/metron/stellar/dsl/functions/RestFunctions.java]: Is this meant to be set to error log level? Wouldn't debug be more appropriate?

**Comment by mmiklavc** [comment]
There may be some users that want a more strict interpretation of the content length. It might make sense to have the default be similar to browsers' quirks mode and be forgiving, per the rfc declaring this as optional, but also allow users to set a strict mode that will throw an error if content length does not equal actual content length. Can we add a config option for this?

**Comment by merrimanr** [comment]
The latest commit adds an optional check for the content length.

File: metron-stellar/stellar-common/src/main/java/org/apache/metron/stellar/dsl/functions/RestFunctions.java (hunk 2 above)
Comment (merrimanr) [reply metron-stellar/stellar-common/src/main/java/org/apache/metron/stellar/dsl/functions/RestFunctions.java]: Yes this was a mistake.  Fixed in latest commit.

**Comment by mmiklavc** [comment]
Thanks for the updates @merrimanr, this looks good. I'm +1 by inspection.

**Comment by merrimanr** [comment]
Thanks @mmiklavc.  Are you good @justinleet and @nickwallen?

**Comment by nickwallen** [comment]
I am happy with the logging.  Thanks for adding that.  Consider me a +0 only because I have not reviewed it all. Please feel free to go ahead based on @mmiklavc 's +1.

**Comment by justinleet** [comment]
+1, this looks good!

**Pull request closed without merging**